synthesizer/process/src/stack/mod.rs

```diff
@@ -81,7 +81,7 @@ pub type Assignments<N> = Arc<RwLock<Vec<(circuit::Assignment<<N as Environment>
 pub enum CallStack<N: Network> {
     Authorize(Vec<Request<N>>, PrivateKey<N>, Authorization<N>),
     Synthesize(Vec<Request<N>>, PrivateKey<N>, Authorization<N>),
-    CheckDeployment(Vec<Request<N>>, PrivateKey<N>, Assignments<N>),
+    CheckDeployment(Vec<Request<N>>, PrivateKey<N>, Assignments<N>, u64),
```

Comment:
There doesn't seem to be final verification that the deployment actually has this specified number of constraints. 

We just know that it has less than this number, because verification didn't halt. That might(?) be enought?

Reply:
There is a check before synthesis, and then subsequently for every constraint that's added, which should be sufficient.